src/internal/error/errors.go

```diff
@@ -148,11 +148,12 @@ func UnknownError(err error) *Error {
 
 // IsErr checks whether the err chain contains error matches the code
 func IsErr(err error, code string) bool {
-	var e *Error
-	if errors.As(err, &e) {
-		return e.Code == code
-	}
-	return false
+	return ErrCode(err) == code
+}
```

Comment:
Seems the modify changes the original logic. If we pass a `errors.New("error")` and `GeneralCode`, the original logic returns false, but the current one returns true

Reply:
errors.New("error") got GeneralCode and render 500 status code, right?

Reply:
+1

Reply:
reverted